docs/spec/v2/log.json

```diff
@@ -76,12 +105,122 @@
+          "number"
+        ],
+        "maxLength": 1024
+      }
+    },
+    "log.level": {
+      "description": "Level represents the severity of the recorded log.",
+      "type": [
+        "null",
+        "string"
+      ],
+      "maxLength": 1024
+    },
+    "log.logger": {
+      "description": "Logger represents the name of the used logger instance.",
+      "type": [
+        "null",
+        "string"
+      ],
+      "maxLength": 1024
+    },
+    "log.origin.file.line": {
+      "description": "OriginFileLine represents the line number in the file containing the sourcecode where the log originated.",
+      "type": [
+        "null",
+        "integer"
+      ]
+    },
+    "log.origin.file.name": {
+      "description": "OriginFileName represents the filename containing the sourcecode where the log originated.",
+      "type": [
+        "null",
+        "string"
+      ],
+      "maxLength": 1024
+    },
+    "log.origin.function": {
+      "description": "OriginFunction represents the function name where the log originated.",
+      "type": [
+        "null",
+        "string"
+      ]
+    },
     "message": {
       "description": "Message logged as part of the log. In case a parameterized message is captured, Message should contain the same information, but with any placeholders being replaced.",
       "type": [
         "null",
         "string"
       ]
+    },
+    "process.thread.name": {
+      "description": "ProcessThreadName represents the name of the thread.",
+      "type": [
+        "null",
+        "string"
+      ],
+      "maxLength": 1024
+    },
+    "service.environment": {
+      "description": "ServiceEnvironment represents the environment the service which originated the log line is running in.",
+      "type": [
+        "null",
+        "string"
+      ],
+      "maxLength": 1024
+    },
+    "service.name": {
+      "description": "ServiceName represents name of the service which originated the log line.",
+      "type": [
+        "null",
+        "string"
+      ],
+      "maxLength": 1024
+    },
+    "service.node.name": {
+      "description": "ServiceNodeName represents a unique node name per host for the service which originated the log line.",
+      "type": [
+        "null",
+        "string"
+      ],
+      "maxLength": 1024
+    },
+    "service.version": {
+      "description": "ServiceVersion represents the version of the service which originated the log line.",
+      "type": [
+        "null",
+        "string"
+      ],
+      "maxLength": 1024
+    },
+    "trace.id": {
+      "description": "TraceID holds the hex encoded 128 random bits ID of the correlated trace.",
+      "type": [
+        "null",
+        "string"
+      ],
+      "maxLength": 1024
+    },
+    "transaction.id": {
+      "description": "TransactionID holds the hex encoded 64 random bits ID of the correlated transaction.",
+      "type": [
+        "null",
+        "string"
+      ],
+      "maxLength": 1024
```

Comment:
Initially, I added it but the [ECS spec](https://github.com/elastic/ecs-logging/blob/main/spec/spec.json) doesn't mention `span.id`. Not sure if any ECS library will use it.

Reply:
Right, I see. [`apmzap`](https://github.com/elastic/apm-agent-go/blob/main/module/apmzap/fields.go) does set it. @felixbarny do you have any concern about allowing `span.id` even if it's not in the ECS logging spec?

Reply:
I know that some agents do add a `span.id` so I think we should add it and consider adding it to the ECS spec. The thing is that currently the `span.id` in logs is not used in the product but that might change in the future. But either way, there's no harm in adding it. Especially since we do want to support dynamic fields going forward.